compiler/ballerina-parser/src/test/resources/statements/do-stmt/do_stmt_source_10.bal

```diff
@@ -0,0 +1,7 @@
+public function test() {
+    do {
+        fail error("error!");
+    } on fail var error(m) {
```

Comment:
Also add tests with error binding patterns where the member binding patterns are other binding patterns. E.g., cause and a detail field use error and a list/mapping binding pattern.